display_list/dl_builder.cc

```diff
@@ -44,10 +65,15 @@ void* DisplayListBuilder::Push(size_t pod, Args&&... args) {
   size_t size = SkAlignPtr(sizeof(T) + pod);
   FML_CHECK(size < (1 << 24));
   if (used_ + size > allocated_) {
-    static_assert(is_power_of_two(DL_BUILDER_PAGE),
+    static_assert(is_power_of_two(kDLBuilderMaxGrowth),
                   "This math needs updating for non-pow2.");
-    // Next greater multiple of DL_BUILDER_PAGE.
-    allocated_ = (used_ + size + DL_BUILDER_PAGE) & ~(DL_BUILDER_PAGE - 1);
+    // Next power of two, up to a limit of DL_BUILDER_MAX_GROWTH. This
+    // assumes that the requested size to allocate is always less than
+    // kDLBuilderMaxGrowth, which given is over a MB is a reasonable
+    // assumption.
+    uint64_t next_size = NextPowerOfTwo(used_ + size);
```

Comment:
used_ starts at 0, as does allocated_ and the first thing we record is something very small, like 8 bytes. So, we do our first growth to 8 bytes, then 16, then ... when our first growth used to be to 4096.

I think we need both a minimum starting size and a maximum growth size for this...

Reply:
Good point, we should set a min size to the page size.